pkg/otbuiltin/commit.go

```diff
@@ -407,6 +412,62 @@ out:
 	return "", generateError(cerr)
 }
 
+// Return the pointer to metadata
+// Metadata is recreated only if need to add Collection ID or refBindings.
+func addRefs(metadata *C.GVariant, collectionStr string, refStr []string) *C.GVariant {
+	builder := C.g_variant_builder_new(C._g_variant_type(C.CString("a{sv}")))
+	defer C.g_variant_builder_unref(builder)
+
+	if strings.Compare(collectionStr, "") == 0 && len(refStr) == 0 {
+		// Do not need to recreate metadata
+		goto out
+	}
+	// Add Ref bindings
+	if len(refStr) != 0 {
+		sort.Strings(refStr)
+		refs := C.g_ptr_array_new()
+		defer C.g_ptr_array_unref(refs)
+		for _, ref := range refStr {
+			refc := C.CString(ref)
+			C.g_ptr_array_add(refs, (C.gpointer)(refc))
+		}
+		refs_v := C.g_variant_new_strv(
+			(**C.gchar)(unsafe.Pointer(refs.pdata)),
+			C.gssize(len(refStr)))
+		C._g_variant_builder_add_twoargs(builder, C.CString("{s@v}"),
+			C.CString("ostree.ref-binding"),
+			C.g_variant_new_variant(refs_v))
+		// C.OSTREE_COMMIT_META_KEY_REF_BINDING is available since ostree 2017.9
+	}
+
+	// Add Collection ID binding
+	if strings.Compare(collectionStr, "") != 0 {
+		collection_id := C.g_variant_new_string((*C.gchar)(C.CString(collectionStr)))
+		C._g_variant_builder_add_twoargs(builder, C.CString("{sv}"),
+			C.CString("ostree.collection-binding"),
+			collection_id)
+		// C.OSTREE_COMMIT_META_KEY_COLLECTION_BINDING is available since ostree 2018.6
```

Comment:
ditto; 2018.6 is new enough as well  (newer versions are available in Debian backports)